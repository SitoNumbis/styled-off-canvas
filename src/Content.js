import React, { useContext } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import Context from './Context'

const Div = styled.div(({ blur, position, duration, isOpen, width }) => `
  filter: ${blur && isOpen ? 'blur(3px)' : 'inherit'};
  height: 100%;
  transform: translate3d(${isOpen ? position === 'left' ? `-${width}` : width : 0}, 0, 0);
  transition: all ${duration} ease 0s;
`)

/**
 * Content wrapper component
 *
 * @param   {string}    className styled-components className for custom styling
 * @returns {component}           React component
 */
const Content = ({ className, children }) => {
  const {
    blur,
    isOpen,
    menuDuration,
    position,
    push,
    width
  } = useContext(Context)

  if (!push) return children

  return (
    <Div
      className={className}
      blur={blur}
      duration={menuDuration}
      position={position === 'left' ? 'right' : 'left'}
      isOpen={isOpen}
      width={width}
    >
      {children}
    </Div>
  )
}

Content.propTypes = {
  className: PropTypes.object
}

export default Content
