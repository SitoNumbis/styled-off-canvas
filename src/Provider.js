import React from 'react'
import PropTypes from 'prop-types'

import Context from './Context'

/**
 * Provider component
 *
 * @param   {boolean}   closeOnEsc        if the menu should close on esc keydown
 * @param   {boolean}   isOpen            if the menu should be visible or not
 * @param   {string}    menuBackground    background color of the menu
 * @param   {string}    menuDuration      duration of the open/close animation for the menu
 * @param   {function}  onClose           callback function if menu closes
 * @param   {string}    overlayBackground background color of the menu
 * @param   {string}    overlayDuration   duration of the open/close animation for the overlay
 * @param   {string}    overlayOpacity    opacity of the overlay
 * @param   {string}    position          position of the menu (left or right)
 * @param   {string}    width             maximum width of the menu
 * @returns {component}                   React component
 */
const Provider = ({
  children,
  closeOnEsc = true,
  isOpen = false,
  menuBackground = '#fff',
  menuDuration = '500ms',
  onClose,
  overlayBackground = '#000',
  overlayDuration = '500ms',
  overlayOpacity = 0.2,
  position = 'right',
  width = '300px'
}) => (
  <Context.Provider
    value={{
      closeOnEsc,
      isOpen,
      menuBackground,
      menuDuration,
      onClose,
      overlayBackground,
      overlayDuration,
      overlayOpacity,
      position,
      width
    }}
  >
    {children}
  </Context.Provider>
)

Provider.propTypes = {
  closeOnEsc: PropTypes.bool,
  isOpen: PropTypes.bool,
  menuBackground: PropTypes.string,
  menuDuration: PropTypes.string,
  onClose: PropTypes.func.isRequired,
  overlayBackground: PropTypes.string,
  overlayDuration: PropTypes.string,
  overlayOpacity: PropTypes.number,
  position: PropTypes.oneOf(['left', 'right']),
  width: PropTypes.string
}

export default Provider
