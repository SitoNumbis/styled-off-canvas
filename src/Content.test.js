import React from 'react'
import renderer from './helper/tests/jest'

import Provider from './Provider'
import Content from './Content'

describe('Content', () => {
  test('default', () => {
    const tree = renderer(<Provider onClose={() => {}} push><Content>content</Content></Provider>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('without push', () => {
    const tree = renderer(<Provider onClose={() => {}}><Content>content</Content></Provider>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('open', () => {
    const tree = renderer(<Provider onClose={() => {}} push isOpen><Content>content</Content></Provider>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('left', () => {
    const tree = renderer(<Provider onClose={() => {}} position='left' push isOpen><Content>content</Content></Provider>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('blur', () => {
    const tree = renderer(<Provider onClose={() => {}} blur push isOpen><Content>content</Content></Provider>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
