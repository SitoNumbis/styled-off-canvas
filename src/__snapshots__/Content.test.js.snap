// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content blur 1`] = `
<div
  className="Content__Div-sc-1e2zm8s-0 eoeqoK"
  width="300px"
>
  content
</div>
`;

exports[`Content default 1`] = `
<div
  className="Content__Div-sc-1e2zm8s-0 ghRgXy"
  width="300px"
>
  content
</div>
`;

exports[`Content left 1`] = `
<div
  className="Content__Div-sc-1e2zm8s-0 hqHScF"
  width="300px"
>
  content
</div>
`;

exports[`Content open 1`] = `
<div
  className="Content__Div-sc-1e2zm8s-0 hQhenM"
  width="300px"
>
  content
</div>
`;

exports[`Content without push 1`] = `"content"`;
